lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -62,9 +92,26 @@ export class RoutingApiQuoter implements Quoter {
     }
   }
 
-  buildRequest(request: ClassicRequest): string {
+  async buildRequest(request: ClassicRequest): Promise<string> {
+    const amount = request.info.amount.toString();
     const tradeType = request.info.type === TradeType.EXACT_INPUT ? 'exactIn' : 'exactOut';
+
+    // exact out needs to calculate the portion amount at this point,
+    // hence more complicated logic
```

Comment:
nit: maybe explain why exact out needs to calculate a flat portion amount compared to exact in